src/main/java/com/sofrelgroup14/ci/controller/BuildController.java

```diff
@@ -114,15 +121,76 @@ public void buildAndCreateLog(@RequestBody String body) {
 
             // TODO (notification): Set commit status depending on success (or not) of build
 
             // Remove repo (-R for directory, -f to skip prompt "are you sure?")
             ProcessBuilder pbRmRepo = new ProcessBuilder("rm", "-R", "-f", "repo");
             pbRmRepo.start().waitFor(); // Start process and block this program thread until process has finished.
+            
+            sendStatus(buildSuccess == true ? "success" : "failure", commitHash, _id.toString());
 
+        } catch (Exception e) {
+            e.printStackTrace();
+            // Server crashed or something :(
+            sendStatus("failure", commitHash, ""); 
+        }
+    }
+
+
+    private void sendStatus(String status, String commitHash, String jobID) {
+        try {
+            // Abort if no access token.
+            if (System.getenv("GH_ACCESS_TOKEN") == null) {
+                System.out.println("NO GH_ACCES_TOKEN DEFINED. WON'T NOTIFY GITHUB ABOUT BUILD STATUS.");
+                return;
+            };
+
+            // https://www.baeldung.com/java-http-request
+            // https://www.baeldung.com/httpurlconnection-post
+            URL url = new URL("https://api.github.com/repos/sofrel-group14/ci/statuses/" + commitHash);
+    
+            HttpURLConnection con = (HttpURLConnection) url.openConnection();
+            con.setDoOutput(true);
+            con.setRequestMethod("POST");
+            con.setRequestProperty("Accept", "application/vnd.github.v3+json");
+            // Generate a token at https://github.com/settings/tokens/new and give it the correct access rights
+            // I think it is workflow/repo
+            con.setRequestProperty("Authorization", "token " + System.getenv("GH_ACCESS_TOKEN"));
+    
+            // Put arguments to JSON body
+            JSONObject body = new JSONObject();
+            // Link to resulting job. Only append if status is not pending or nonzero id.
+            // TODO: Make linking to pretty frontend possible
+            if (!jobID.equals("pending") && jobID.length() != 0) {
+                body.put("target_url", "http://axelelmarsson.se/logs/" + jobID);
+            }
+            body.put("description", "Our CI Server");
+            // Job status
+            body.put("state", status);
+    
+            // Send body
+            DataOutputStream os = new DataOutputStream(con.getOutputStream());
+            byte[] input = body.toString().getBytes("utf-8");
+            System.out.println(con);
+            System.out.println(body.toString());
+            os.write(input, 0, input.length);
+            os.close();
+            
+            BufferedReader in = new BufferedReader(
+                                    new InputStreamReader(
+                                        con.getInputStream()));
+            String decodedString;
+            while ((decodedString = in.readLine()) != null) {
+                System.out.println(decodedString);
+            }
```

Comment:
What is printed here? Is it the response from GitHub? Overall I think it would be good if all of our `println`:s had a description above them, i.e.

```java
String decodedString;
while ((decodedString = in.readLine()) != null) {
    System.out.println("\nGitHub response when setting commit status");
    System.out.println(decodedString);
}
```
or something like that. This would make it easier to scan the debug output if something goes wrong 👍